lib/plugins/physics.js

```diff
@@ -35,7 +35,8 @@ function inject (bot) {
   let lastSentYaw = null
   let doPhysicsTimer = null
   let lastPhysicsFrameTime = null
-  bot.physicEnabled = false
+  let shouldUsePhysics = false
```

Comment:
why did you introduce shouldUsePhysics ? it's duplicated with bot.physicsEnabled

Reply:
Before this commit, physicsEnabled represendet both the internal state of the physics plugin, that disabled physics when riding a horse, using a boat etc; but at the same time was also the exposed property. This meant, that whenever the player got a position(teleport) packet from the server (which is pretty common), or dismounted a boat/horse, physicsEnabled would reset to true.
By introducing shouldUsePhysics for the internal state and only using physics when both the internal state and the exposed toggle are true, physics will only be enabled when they are usable (e.g. not riding), and also not explicitly disabled by the user. See further down, where physicsEnabled was always set to true when the bot got a position packet; this (among other occourances) now sets shouldUsePhysics instead.